sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PortEvaluationStage.java

```diff
@@ -262,10 +268,25 @@ protected void setProtos(PodInfoBuilder podInfoBuilder, Protos.Resource resource
             .get(Constants.PORTS_RESOURCE_TYPE);
     Optional<Integer> dynamicPort = Optional.empty();
     if (availablePorts != null) {
-      dynamicPort = availablePorts.getRanges().getRangeList().stream()
-          .flatMap(r -> IntStream.rangeClosed((int) r.getBegin(), (int) r.getEnd()).boxed())
-          .filter(p -> !consumedPorts.contains(p))
-          .findFirst();
+      if (!spec.getRanges().isEmpty()) {
+        List<Integer> constrainedPorts = new ArrayList<>();
+        for (RangeSpec range : spec.getRanges()) {
+          constrainedPorts.addAll(
+              IntStream.range(range.getBegin(), range.getEnd())
+                  .boxed()
+                  .collect(Collectors.toList()));
+        }
+        dynamicPort = availablePorts.getRanges().getRangeList().stream()
+            .flatMap(r -> IntStream.rangeClosed((int) r.getBegin(), (int) r.getEnd()).boxed())
```

Comment:
I'm not a fan of single letter variables (with the exception of i and the like for indexes)
I assume r == range and p == port
I love terse... but not at the loss of readability

Reply:
resolving since this is a convention already followed in dcos-commons